spec.emu

```diff
@@ -132,6 +920,9 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
             1. <ins>If _moduleSourceObject_ is *null*, then</ins>
               1. <ins>Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; a newly created *TypeError* object &raquo;).</ins>
               1. <ins>Return ~unused~.</ins>
+            1. <ins>Assert: _moduleSourceObject_.[[Prototype]] is %AbstractModuleSourcePrototype%.</ins>
+            1. <ins>Assert: _moduleSourceObject_ has a [[HostSourceData]] internal slot.</ins>
+            1. <ins>Assert: _moduleSourceObject_ is a ModuleSource instance if and only if _module_ is a Source Text Module Record.</ins>
```

Comment:
This assertion is only valid if we enforce it somewhere (assertions are non-normative, they only assert existing internal properties of the spec). We could do it in the description of Module Record's [[ModuleSourceObject]] internal slot, or in the requirements of the HostLoadImportedModule hook (the former is better, because the hook is not called for the top-level module).

Reply:
See updated assertions and description.